crates/sui-framework/sources/crypto/hash.move

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Mysten Labs, Inc.
```

Comment:
do we want this to be combined with digest.move? 

also i think its worth adding in comments a link to where to find sha256 since its offered in stdlib and ppl may come here first.

Reply:
> do we want this to be combined with digest.move? that class only has a wrapper around a digest struct, but the similarity in name can be confusing.

We could do that. As far as I can see, digest.move is just a wrapper around a byte vector, ensuring that the size is correct. Here I followed the api of the hash functions from the standard lib which just returned a byte vector.

Reply:
I've given it a bit of thought, and I don't think we should combine this module with digest.move. I've used the same module name (`hash`) as in the std lib, so if anything, the contents of digest.move should be added to the hash module, but I also think it's fine to keep them apart, since `digest` here is more about the output of hash functions.

I've added a note in the doc about where to find the stdlib hash functions, and we could also add a note in digest.move to avoid confusion?

Reply:
makes sense to me to include comment!